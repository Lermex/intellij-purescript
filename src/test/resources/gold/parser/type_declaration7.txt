Purescript File
  PSProgramImpl(Program)
    PSModuleImpl(Module)
      PsiElement(module)('module')
      PsiElement(whitespace)(' ')
      PSProperNameImpl(ModuleName)
        PSProperNameImpl(Qualified)
          PsiElement(proper name)('Main')
        PsiElement(whitespace)(' ')
      PsiElement(where)('where')
      PsiElement(whitespace)('\n\n')
      PSTypeSynonymDeclarationImpl(TypeSynonymDeclaration)
        PsiElement(type)('type')
        PsiElement(whitespace)(' ')
        PSIdentifierImpl(TypeConstructor)
          PsiElement(proper name)('RandomConsoleEffect')
          PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSTypeImpl(Type)
          PSTypeAtomImpl(TypeAtom)
            PSIdentifierImpl(TypeConstructor)
              PSProperNameImpl(Qualified)
                PSProperNameImpl(ProperName)
                  PsiElement(proper name)('RandomConsoleEffects')
                  PsiElement(whitespace)(' ')
          PSTypeAtomImpl(TypeAtom)
            PsiElement(()('(')
            PSRowImpl(Row)
              <empty list>
            PsiElement())(')')