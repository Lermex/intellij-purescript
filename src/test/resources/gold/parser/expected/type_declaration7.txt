Purescript File
  PSModuleDeclarationImpl(MODULE_DECLARATION)
    PsiElement(PSTokenType.module)('module')
    PSModuleNameImpl(MODULE_NAME)
      PsiElement(PSTokenType.PROPER_NAME)('Main')
    PsiElement(PSTokenType.where)('where')
  PSDeclarationsImpl(DECLARATIONS)
    PSDeclarationImpl(DECLARATION)
      PSTypeDeclarationImpl(TYPE_DECLARATION)
        PsiElement(PSTokenType.type)('type')
        PSDeclarationRefTypeListImpl(DECLARATION_REF_TYPE_LIST)
          PSDeclarationRefTypeImpl(DECLARATION_REF_TYPE)
            PsiElement(PSTokenType.PROPER_NAME)('RandomConsoleEffect')
        PsiElement(PSTokenType.=)('=')
        PSParenthesisedDeclarationRefTypeListImpl(PARENTHESISED_DECLARATION_REF_TYPE_LIST)
          PSDeclarationRefTypeImpl(DECLARATION_REF_TYPE)
            PsiElement(PSTokenType.PROPER_NAME)('RandomConsoleEffects')
        PsiElement(PSTokenType.()('(')
        PsiElement(PSTokenType.))(')')